### PR TITLE
HPCC-26750 Fix issue with instanceName not being consumed in daserver startup

### PR DESCRIPTION
**Pull request by Michael-Gardner**
Signed-off-by: Michael Gardner <michael.gardner@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by Michael-Gardner** [comment]
@ghalliday When we consolidated the daemon code back in 8.2 this error snuck in.  Originally (in daserver.cpp) we used ++i to bump up which argv we were referencing in the original for loop.  This consumed the expected instanceName argument after --daemon|-d properly. When we moved the code over into a function in jutil, we were passing the args by value and the original for loop in daserver.cpp had no idea we had used the instanceName already to write a pidfile.

This change bumps up i in the for loop when we find --daemon|-d and we no longer hit an 'unknown' argument and just print out the usage.

I've gone through the other 13 references for checkCreateDaemon and ensured this issue isn't anywhere else in the branch.

File: system/jlib/jutil.cpp [hunk 1]
```diff
@@ -3282,7 +3282,8 @@ bool checkCreateDaemon(unsigned argc, const char * * argv)
 #ifndef _CONTAINERIZED
     for (unsigned i=0;i<(unsigned)argc;i++) {
         if (streq(argv[i],"--daemon") || streq(argv[i],"-d")) {
-            if (daemon(1,0) || write_pidfile(argv[++i])) {
```
Comment (ghalliday) [inline system/jlib/jutil.cpp:3285]: Why the extra check (seems a reasonable check), but if you are improving the error checking it should also be checking that i+1 < argc

**Review by ghalliday — APPROVED** [review]
@Michael-Gardner change looks good.  One comment related to your extra checking, but could merge as is.

File: system/jlib/jutil.cpp (hunk 1 above)
Comment (Michael-Gardner) [reply system/jlib/jutil.cpp:3285]: I'm just checking that the argument after --daemon isn't another flag.  You're absolutely right, I should add a check for i + 1 < argc.

Comment (ghalliday) [reply system/jlib/jutil.cpp:3285]: With the extra check you are preventing the filename from starting with a -, but you should also probably have an equivalent check in daserver.cpp.  It seems an unnecessary extra test.

**Comment by Michael-Gardner** [comment]
Closing to use @lpezet PR's instead

**Comment by Michael-Gardner** [comment]
Reopening and removing the jutil changes. Just doing the increment in daserver.

**Pull request closed without merging**